forc-plugins/forc-doc/src/render.rs

```diff
@@ -620,72 +603,57 @@ impl Renderable for AllDoc {
+    fn sidebar(&self) -> Sidebar;
+}
+/// Sidebar component for quick navigation.
+struct Sidebar {
+    module_info: ModuleInfo,
+    href_path: String,
 }
-/// Sidebar component
-fn sidebar(
-    module_depth: usize,
-    location: String,
-    href: String, /* TODO: sidebar_items */
-) -> Box<dyn RenderBox> {
-    let mut logo_path = module_depth_to_path_prefix(module_depth);
-    logo_path.push_str("assets/sway-logo.svg");
+impl Renderable for Sidebar {
+    fn render(self) -> Box<dyn RenderBox> {
+        let logo_path = self
+            .module_info
+            .to_html_shorthand_path_str("assets/sway-logo.svg");
 
-    box_html! {
-        nav(class="sidebar") {
-            a(class="sidebar-logo", href=href) {
-                div(class="logo-container") {
-                    img(class="sway-logo", src=logo_path, alt="logo");
+        box_html! {
+            nav(class="sidebar") {
+                a(class="sidebar-logo", href=&self.href_path) {
+                    div(class="logo-container") {
+                        img(class="sway-logo", src=logo_path, alt="logo");
+                    }
                 }
-            }
-            h2(class="location") {
-                a(href="#") { : location; }
-            }
-            div(class="sidebar-elems") {
-                section {
-                    // TODO: add connections between item contents and
-                    // sidebar nav. This will be dynamic e.g. "Variants"
-                    // for Enum, and "Fields" for Structs
+                h2(class="location") {
+                    a(href="#") { : self.module_info.location(); }
+                }
+                div(class="sidebar-elems") {
+                    section {
+                        // TODO: add connections between item contents and
+                        // sidebar nav. This will be dynamic e.g. "Variants"
+                        // for Enum, and "Fields" for Structs, and also will be different
+                        // based on the type of section e.g. Index, All or Item.
+                    }
                 }
             }
         }
     }
 }
-/// Creates a String version of the path to an item, used in navigating from
-/// all.html to items.
-fn qualified_file_path(module_prefix: &Vec<String>, file_name: String) -> String {
-    let mut file_path = PathBuf::new();
-    for prefix in module_prefix {
-        if prefix != &module_prefix[0] {
-            file_path.push(prefix)
-        }
-    }
-    file_path.push(file_name);
-
-    file_path.to_str().unwrap().to_string()
-}
-/// Create a path prefix string for navigation from the `module_depth`
-fn module_depth_to_path_prefix(module_depth: usize) -> String {
-    (1..module_depth).map(|_| "../").collect::<String>()
-}
 /// Creates an HTML String from an [AttributesMap]
-pub(crate) fn attrsmap_to_html_string(attributes: &AttributesMap) -> String {
+pub(crate) fn attrsmap_to_html_str(attributes: AttributesMap) -> String {
```

Comment:
Normally we use the name `str` if talking about the `str` slice type. I think the original name ending with `_string` is probably better as it's clearer that we return a `String` and not a `str`.

Reply:
Ah, another part where I had changed the return type to slice when i was refactoring but forgot to change it back.